src/patterns.rs

```diff
@@ -259,56 +259,52 @@ fn rewrite_tuple_pat(pats: &[ptr::P<ast::Pat>],
         pat_vec.insert(pos, dotdot);
     }
 
-    if pat_vec.is_empty() {
-        path_str
```

Comment:
Yup, that worked; I thought I had tried this before, but had inadvertently moved the check to the beginning of the function, which meant it failed on tuples with only `..`.